src/pages/smart-collection/store/sagas.ts

```diff
@@ -28,17 +27,15 @@ import { fetchSmartCollection, fetchSmartCollectionSucceeded } from './slice'
 const COLLECTIONS_LIMIT = 25
 
 function* fetchHeavyRotation() {
-  const topListens = yield call(Explore.getTopUserListens)
+  const topListens = yield* call(Explore.getTopUserListens)
 
-  const users = yield call(
+  const users = yield* call(
     retrieveUsers,
-    topListens.map((t: any) => t.userId)
+    topListens.map(t => t.userId)
   )
   const trackIds = topListens
-    .filter(
-      (track: any) => !(users.entries[track.userId]?.is_deactivated ?? true)
-    )
-    .map((listen: any) => ({
+    .filter(track => !(users.entries[track.userId]?.is_deactivated ?? true))
```

Comment:
this logic is a lil wild :D so essentially filter out tracks that dont have user id, or the user is deactivated? i think it might be a little cleaner to do an if statement checking user exists first, what do you think?

Reply:
yeah def could clean it up now that it's not hotfixing time.